src/collective/documentgenerator/browser/views.py

```diff
@@ -117,13 +129,233 @@ def render(self):
+                                self.sub_templates[sub_template_uid]
+                            )
+
+                if (
+                    hasattr(pod_template, "mailing_loop_template")
+                    and pod_template.mailing_loop_template
+                ):
+                    self.check_mailing_loop(pod_template, obj, output_format)
+
+    def check_pod_template(self, pod_template, obj, output_format):
+        try:
+            view = obj.restrictedTraverse("@@document-generation")
+            view()
+            view._generate_doc(
+                pod_template, output_format=output_format, raiseOnError=True
+            )
+            self.clean.append((pod_template, obj))
+
+        except Exception as exc:
+            self.error.append((pod_template, obj, (_("Error"), str(exc))))
+        self.left_to_verify.remove(pod_template)
+
+    def check_mailing_loop(self, mailed_template, obj, output_format):
+        folder = api.content.create(
+            type="Folder", title=u"Folder", id="temp_folder", container=self.context
+        )
+
+        def do_nothing(ignored_param):
+            pass
+
+        mailing_loop_template_uid = mailed_template.mailing_loop_template
+        try:
+            if mailing_loop_template_uid not in self.mailing_loop_templates.keys():
+                mailing_loop_template = uuidToObject(mailing_loop_template_uid)
+                self.mailing_loop_templates[
+                    mailing_loop_template_uid
+                ] = mailing_loop_template
+            generation_view = folder.restrictedTraverse(
+                "@@persistent-document-generation"
+            )
+            generation_view.redirects = do_nothing
+            generation_view(mailed_template.UID(), "odt")
+            persistant_doc = folder.listFolderContents()[0]
+            view = folder.restrictedTraverse(
+                "@@mailing-loop-persistent-document-generation"
+            )
+            view.redirects = do_nothing
+            view(document_uid=persistant_doc.UID())
+            # double check anyway just in case there is an error inside the result file
+            view._generate_doc(
+                self.mailing_loop_templates[mailing_loop_template_uid],
+                output_format=output_format,
+                raiseOnError=True,
+            )
+
+            self.clean.append(
+                (self.mailing_loop_templates[mailing_loop_template_uid], obj)
+            )
+        except Exception as exc:
+            self.error.append(
+                (
+                    self.mailing_loop_templates[mailing_loop_template_uid],
+                    obj,
+                    (_("Error"), str(exc)),
+                )
+            )
+        finally:
+            api.content.delete(obj=folder)
+        self.left_to_verify.remove(
+            self.mailing_loop_templates[mailing_loop_template_uid]
+        )
+
+    def find_pod_templates(self):
```

Comment:
L'idée dans l'intégration dans PM est de continuer à pouvoir le lancer depuis la MeetingConfig donc on devra juste surcharger find_pod_templates

Reply:
Pour la 4.2 il faudra pouvoir le lancer de manière globale aussi.
Ce sera probablement depuis l'interface de search and replace mais ...

Reply:
Donc surecharger cette méthode ça ne me semble pas bon.
je vais ajouter un arg au call afin de lui préciser un scope et par défaut ce sera tt le site